go/vt/vtgate/vtgate.go

```diff
@@ -1074,3 +1074,49 @@ func annotateBoundShardQueriesAsUnfriendly(queries []*vtgatepb.BoundShardQuery)
 		queries[i].Query.Sql = sqlannotation.AnnotateIfDML(q.Query.Sql, nil)
 	}
 }
+
+// unambiguousKeyspaceBKSIQ is a helper function used in the
+// ExecuteBatchKeyspaceIds method to determine the "keyspace" label for the
+// stats reporting.
+// If all queries target the same keyspace, it returns that keyspace.
+// Otherwise it returns an empty string.
+func unambiguousKeyspaceBKSIQ(queries []*vtgatepb.BoundKeyspaceIdQuery) string {
+	if len(queries) == 0 {
+		return ""
+	}
+
+	if len(queries) == 1 {
+		return queries[0].Keyspace
+	}
+
+	keyspace := queries[0].Keyspace
+	for _, q := range queries {
```

Comment:
range queries[1:] ?

Reply:
Thanks! Checking the first element again bothered me as well but I couldn't think of a way to skip it while still using the range syntax. So I totally forgot about this :)